app/webpacker/controllers/product_actions_controller.js

```diff
@@ -0,0 +1,18 @@
+import ApplicationController from "./application_controller";
+
+export default class extends ApplicationController {
+  static values = { id: Number };
+
+  setDeleteModalDataSet(event) {
+    try {
+      const modalId = this.element.dataset.modalLinkTargetValue; // whether variant or product delete modal
+      const deleteButtonQuery = `#${modalId} #modal-confirm-button`;
+      const deleteButton = document.querySelector(deleteButtonQuery);
+      deleteButton.setAttribute("data-current-id", this.idValue);
+    } catch (e) {
+      // In case of any type of error in setting the dataset value, stop the further actions i.e. opening the modal
+      event.stopImmediatePropagation();
+      throw e;
+    }
+  }
+}
```

Comment:
- Let me explain the motive of this action here:
  - As we are just [using a single modal for each product and variant](https://github.com/openfoodfoundation/openfoodnetwork/pull/11846#discussion_r1435620397), we want to dynamically set the reflex dataset params.
  - We need to make sure the correct reflex params (_for now, it's just the currently selected product/variant id_) are set before we can call the reflex action on the confirmation button
  - This makes sure that we delete the selected product/variant by just using a single modal each.
- Suggested ideas sound good. I now see that this logic can be used generically.  However I've another idea where we can:
  - Move this method to the `modal_link_controller`.
  - Make the method generic enough to be used with any modal on any page
  - For any modal to be opened, we have to use the `modal_link_controller`  . So it seems a perfect fit to be placed here because we can use this action wherever we want to set some reflex/action params on the modal whose link is being clicked to be opened.
 
Above idea has been implemented [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/11846/commits/17bd3bb0cc0d7201a40a4165b571837a2acd35e9). Would love to hear your thoughts on it. Thanks :)

Reply:
That looks good, thanks for trying that out! Now there's one less controller to manage 👍